networkx/readwrite/json_graph/node_link.py

```diff
@@ -80,22 +110,43 @@ def node_link_data(G, attrs=None):
 
     Attribute 'key' is only used for multigraphs.
 
+    To use `node_link_data` in conjunction with `node_link_graph`,
+    the keyword names for the attributes must match.
+
     See Also
     --------
     node_link_graph, adjacency_data, tree_data
     """
+    # ------ TODO: Remove between the lines after signature change is complete ----- #
+    if attrs is not None:
```

Comment:
Maybe it's an overkill but should there be a check here if there is a mismatch between the `attrs` attribute dict and the keyword args? Something like could be a bit confusing 
``` python
>>> data1 = nx.node_link_data(G, attrs={'source':'s'}, source='t')
>>> data1
{'directed': False,
 'multigraph': False,
 'graph': {},
 'nodes': [{'name': 'A'}, {'name': 'B'}],
 'links': [{'s': 'A', 'target': 'B'}]}

```

This is probably an overkill as `attrs` would be visibly deprecated if someone tries to use the other keyword arguments

Reply:
What if I documented the behavior by adding this line to the docstring...

> Until the `attrs=` keyword is removed, it takes precedence over the new keywords, should they both be used in the same call (not recommended).

Reply:
Sounds good to me :)